ff_dir.c

```diff
@@ -2941,160 +2949,165 @@ FF_Error_t FF_CreateDirent( FF_IOManager_t * pxIOManager,
     /* Round-up the number of LFN's needed: */
     xLFNCount = ( BaseType_t ) ( ( NameLen + 12 ) / 13 );
 
-    #if ( ffconfigUNICODE_UTF16_SUPPORT != 0 )
-    {
-        FF_MakeNameCompliant( pxDirEntry->pcFileName ); /* Ensure we don't break the Dir tables. */
-    }
-    #else
-    {
-        FF_MakeNameCompliant( pxDirEntry->pcFileName ); /* Ensure we don't break the Dir tables. */
-    }
-    #endif
-    memset( pucEntryBuffer, 0, sizeof( pucEntryBuffer ) );
-
-    #if ( ffconfigLFN_SUPPORT != 0 )
-    {
-        /* Create and push the LFN's. */
-        /* Find enough places for the LFNs and the ShortName. */
-        xEntryCount = xLFNCount + 1;
-    }
-    #else
-    {
-        xEntryCount = 1;
-    }
-    #endif
-
-    /* Create the ShortName. */
-    FF_LockDirectory( pxIOManager );
-
-    do
+    if( FF_MakeNameCompliant( pxDirEntry->pcFileName ) )
```

Comment:
The checker for MISRA rules will prefer to see a comparison here:
~~~diff
-    if( FF_MakeNameCompliant( pxDirEntry->pcFileName ) )
+    if( FF_MakeNameCompliant( pxDirEntry->pcFileName ) == pdTRUE )
~~~

Reply:
Use the ErrorCode names not pdTRUE.

```suggestion
    if( FF_MakeNameCompliant( pxDirEntry->pcFileName )  == FF_ERR_NONE)
```